### PR TITLE
Move pre commit to push

### PR DESCRIPTION
**Pull request by MichelML**
move pre commit to pre push to save time, and make the build fail if pretty typescript or the build itself fail

### CODE REVIEW

File: src/components/dropdownSearch/DropdownSearch.tsx [hunk 1]
```diff
@@ -99,8 +99,8 @@ export class DropdownSearch extends React.Component<IDropdownSearchProps, void>
           : value;
 
         return <li key={option.value}
-                   className={liClasses}
-                   title={value}>
+          className={liClasses}
+          title={value}>
```
Comment (dcarmichael1) [inline src/components/dropdownSearch/DropdownSearch.tsx]: I changed this part on my PR! I also added Parenthesis around JSX object. You should revert those two lines because they are not related to this PR.

File: package.json [hunk 2]
```diff
@@ -105,6 +105,7 @@
     "node-sass": "4.5.3",
     "postcss-loader": "2.0.6",
     "pre-commit": "1.2.2",
```
Comment (dcarmichael1) [inline package.json]: should we remove this dependency?

File: gulpfile.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ gulp.task('clean', 'Clean all', ['clean:dist', 'clean:docs', 'clean:tests']);
 // <editor-fold desc="Prettify">
 const prettify = (srcPaths, destPath) =>
   gulp.src(srcPaths)
-    .pipe(prettyTypescript())
+    .pipe(prettyTypescript({ emitError: true }))
```
Comment (dcarmichael1) [inline gulpfile.js]: what does that do?

Comment (MichelML) [inline gulpfile.js]: failing if finding errors

Comment (MichelML) [inline gulpfile.js]: https://github.com/coveo/pretty-typescript

File: package.json (hunk 2 above)
Comment (MichelML) [inline package.json]: we could

File: gulpfile.js (hunk 3 above)
Comment (dcarmichael1) [inline gulpfile.js]: Is there any user feedback so we now why we cannot push?

Comment (MichelML) [inline gulpfile.js]: yes

Comment (MichelML) [inline gulpfile.js]: example :

Comment (MichelML) [inline gulpfile.js]: screen shot 2017-08-01 at 1 40 15 pm

Comment (dcarmichael1) [inline gulpfile.js]: nice!
